test/test_images/multicontainer/servingcontainer/servingcontainer.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"knative.dev/serving/test"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	log.Println("serving container received a request.")
+	res, err := http.Get("http://127.0.0.1:8882")
+	if err != nil {
+		log.Fatal(err)
+	}
+	resp, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Fprintln(w, string(resp))
+}
+
+func main() {
```

Comment:
It might be easier to have a single container and just behave differently based on env vars.

ie. you know the serving container will have a $PORT variable

Reply:
chatted with @savitaashture - she suggested having two separate images would better a better representation of how the feature would be used.

resolving